app/main/posts/detail/post-detail-data.directive.js

```diff
@@ -226,5 +229,19 @@ function PostDetailDataController(
                 Notify.apiErrors(errorResponse);
             });
     };
+
+    $scope.close = function () {
```

Comment:
So I changed approach a bit. Instead of trying to call the parent directive function, we just try to use whatever the previous state was.